typewriter/fixes/fixer_utils.py

```diff
@@ -105,6 +211,118 @@ def is_import_stmt(node):
     root.insert_child(insert_pos, Node(syms.simple_stmt, children))
 
 
+# -------------- TYPE_CHECKING LOGIC ------------------ #
+
+
+def _new_type_check_with_import(package, name, root, insert_pos):
+    # type: (Optional[str], str, Node, int) -> None
+    """
+    Inserts a new TYPE_CHECKING block containing a new import statement for package and name
+
+    Parameters
+    -----------
+    package : Optional[str]
+    name : str
+    root : Node
+    insert_pos : int
+    """
+    # [Grammar]
+    # if_stmt: 'if' namedexpr_test ':' suite ('elif' namedexpr_test ':' suite)* ['else' ':' suite]
+    type_check_node = Node(syms.if_stmt,
+                           [Leaf(1, 'if'),
+                            Node(syms.power, [Leaf(1, 'typing'),
```

Comment:
Why `syms.power`? Shouldn't this be `atom_expr`?

Reply:
Copying over an exchange on Slack, for easier future ref:

@ezragoss :
>  i do at least have a question about your atom_expr note. Where in the grammar do you see that this pattern should degrade to that. from what I can tell, the and_expr degrades to power which includes an atom as one of its components
> but
> the structure of that node in the relevant piece of code was me imitating the parsed lib2to3 tree of the expected output
> so my understanding of what is going on is deductive and not constructive

@pmolodo :
> Sorry I missed this earlier - was looking at the power / atom_expr / atom grammar thing though - it looks like I was looking at the grammar for [3.7](https://docs.python.org/3.7/reference/grammar.html), which differs here than that for [2.7](https://docs.python.org/2.7/reference/grammar.html) (and it seems this bit changes again in [3.9](https://docs.python.org/3.9/reference/grammar.html)).
So - yeah, looks like you're right, power is the right Node for [2.7](https://docs.python.org/2.7/reference/grammar.html):

```
power: atom trailer* ['**' factor]
```

In  [3.7](https://docs.python.org/3.7/reference/grammar.html)/[3.8](https://docs.python.org/3.8/reference/grammar.html), though, looks like the trailer is in the atom_expr:

```
atom_expr: ['await'] atom trailer*
```

...and then in [3.9](https://docs.python.org/3.9/reference/grammar.html)/[3.10](https://docs.python.org/3.10/reference/grammar.html), we need a primary:

```
primary:
    | invalid_primary  # must be before 'primay genexp' because of invalid_genexp
    | primary '.' NAME 
    | primary genexp 
    | primary '(' [arguments] ')' 
    | primary '[' slices ']' 
    | atom
```

We can probably ignore [3.9](https://docs.python.org/3.9/reference/grammar.html)/[3.10](https://docs.python.org/3.10/reference/grammar.html) for now, since we have no idea if/when VFX will start using that, and supposedly they're [switching to a new parser](https://www.python.org/dev/peps/pep-0617) anyway, which lib2to3 doesn't really support.
If we just want this to be a one-shot tool, to allow us to upgrade to 3.7, then we might be able to ignore 3.7 too - but if we want to use it in the future, we should probably add a 3.7 switch...